code/game/machinery/telecomms/presets.dm

```diff
@@ -179,6 +179,88 @@
 	freq_listening = list(COLONY_FREQ)
 	faction_shorthand = "colony"
 
+GLOBAL_LIST_EMPTY(all_static_telecomms_towers)
+
+/obj/structure/machinery/telecomms/relay/preset/tower/Initialize()
+	GLOB.all_static_telecomms_towers += src
+	. = ..()
+
+/obj/structure/machinery/telecomms/relay/preset/tower/Destroy()
+	GLOB.all_static_telecomms_towers -= src
+	. = ..()
+
+/obj/structure/machinery/telecomms/relay/preset/tower/mapcomms
+	name = "TC-3T static telecommunications tower"
+	desc = "A static heavy-duty TC-3T telecommunications tower. Used to set up subspace communications lines between planetary and extra-planetary locations. Will need to have extra communication frequencies programmed into it by multitool."
+	use_power = 0
+	idle_power_usage = 10000
+	icon = 'icons/obj/structures/machinery/comm_tower3.dmi'
+	icon_state = "static1"
+	toggled = FALSE
+	bound_height = 64
+	bound_width = 64
+	freq_listening = list(COLONY_FREQ)
+	var/toggle_cooldown = 0
+
+/obj/structure/machinery/telecomms/relay/preset/tower/mapcomms/attack_hand(mob/user)
+	if(toggle_cooldown > world.time) //cooldown only to prevent spam toggling
+		to_chat(user, SPAN_WARNING("\The [src]'s processors are still cooling! Wait before trying to flip the switch again."))
+		return
+	if(!do_after(user, 20, INTERRUPT_NO_NEEDHAND|BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY, src))
+		return
```

Comment:
```suggestion
	var/current_state = on
	if(!do_after(user, 20, INTERRUPT_NO_NEEDHAND|BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY, src))
		return
	if(current_state != on)
		to_chat(user, SPAN_NOTICE("\The [src] is already turned [on ? "on" : "off"]!"))
		return
```
So that if two users are trying to turn on about simultaneously, then after first user turns it on, second user's attempt at turning on does not result in turning off.